scripts/fuzz_opt.py

```diff
@@ -655,7 +655,16 @@ def write_commands(commands, filename):
 
 opt_choices = [
     [],
-    ['-O1'], ['-O2'], ['-O3'], ['-O4'], ['-Os'], ['-Oz'],
+    # Even though -O0 and -O1 by themselves do not generate stack IR, the
+    # optimization level may be overridden by later optimization flags so we
+    # have to be defensive and add round trips for them as well.
```

Comment:
The way optimization option parsing works, each instance of an `-O...` flag adds a marker to the pass sequence where a set of default optimization passes should be inserted and also overrides the current shrink and optimization levels. After all the options are parsed, the markers are replaced with the optimization passes for the last set optimization and shrink levels, so `-O1 -O3` is identical to `-03 -03`.

All other passes except `--generate-stack-ir` are safe in the presence of these overrides because they do not end with stack IR under any optimization or shrink levels.

Reply:
> Like other passes, `-O0` or `-O1` do not end with stack IR because they don't contain `--generate-stack-ir`, no?

`-O0` and `-O1` *do* contain --generate-stack-ir when some later option like `-O3` overrides the shrink and opt levels.

Reply:
Oh, I didn't know that parsing rule...